pyrobosim/test/utils/test_pose_utils.py

```diff
@@ -239,18 +289,57 @@ def test_get_angular_distance():
     assert pose1.get_angular_distance(pose2) == pytest.approx(-3 * np.pi / 4)
 
 
-def test_get_matrices():
-    """Tests getting matrices from a pose."""
-    pose = Pose(x=1.0, y=2.0, z=3.0, roll=np.pi / 2, pitch=0.0, yaw=-np.pi / 2)
-
-    expected_tform = np.array(
-        [
-            [0.0, 1.0, 0.0, 1.0],
-            [0.0, 0.0, -1.0, 2.0],
-            [-1.0, 0.0, 0.0, 3.0],
-            [0.0, 0.0, 0.0, 1.0],
-        ]
+@pytest.mark.parametrize(
+    "x, y, z, roll, pitch, yaw, angle_units, expected_tform",
+    [
+        (
+            1.0,
+            2.0,
+            3.0,
+            np.pi / 2,
+            0.0,
+            -np.pi / 2,
+            "radians",
+            np.array(
+                [
+                    [0.0, 1.0, 0.0, 1.0],
+                    [0.0, 0.0, -1.0, 2.0],
+                    [-1.0, 0.0, 0.0, 3.0],
+                    [0.0, 0.0, 0.0, 1.0],
+                ]
+            ),
+        ),
+        (
+            1.0,
+            2.0,
+            3.0,
+            90.0,
+            0.0,
+            -90.0,
+            "degrees",
+            np.array(
+                [
+                    [0.0, 1.0, 0.0, 1.0],
+                    [0.0, 0.0, -1.0, 2.0],
+                    [-1.0, 0.0, 0.0, 3.0],
+                    [0.0, 0.0, 0.0, 1.0],
+                ]
+            ),
+        ),
+    ],
+)
+def test_get_matrices(x, y, z, roll, pitch, yaw, angle_units, expected_tform):
+    """Test getting matrices from a pose with different angle units"""
+    pose = Pose(
+        x=x,
+        y=y,
+        z=z,
+        roll=roll,
+        pitch=pitch,
+        yaw=yaw,
+        angle_units=angle_units,
     )
+
     expected_translation_matrix = np.eye(4)
     expected_translation_matrix[:3, 3] = expected_tform[:3, 3]
```

Comment:
here as well, you can save yourself a lot of lines by just having the expected transform in the body of the function